espaloma/graphs/deploy.py

```diff
@@ -0,0 +1,171 @@
+    # }
+
+    sys = ff.create_openmm_system(g.mol.to_topology())
+
+    for force in sys.getForces():
+        name = force.__class__.__name__
+        if 'HarmonicBondForce' in name:
+            for idx in range(force.getNumBonds()):
+                idx0, idx1, eq, k = force.getBondParameters(idx)
+                position = bond_lookup[(idx0, idx1)]
+                _eq = g.nodes['n2'].data['eq%s' % suffix][position].detach(
+                    ).numpy().item()
+                _k = g.nodes['n2'].data['k%s' % suffix][position].detach(
+                    ).numpy().item()
+
+                _eq = Quantity(
+                    _eq,
+                    esp.units.DISTANCE_UNIT,
+                ).value_in_unit(OPENMM_BOND_EQ_UNIT)
+
+                _k = 2.0 * Quantity(
+                    _k,
+                    esp.units.FORCE_CONSTANT_UNIT,
+                ).value_in_unit(OPENMM_BOND_K_UNIT)
+
+                force.setBondParameters(idx, idx0, idx1, _eq, _k)
+
+        if 'HarmonicAngleForce' in name:
+            for idx in range(force.getNumAngles()):
+                idx0, idx1, idx2, eq, k = force.getAngleParameters(idx)
+                position = angle_lookup[(idx0, idx1, idx2)]
+                _eq = g.nodes['n3'].data['eq%s' % suffix][position].detach(
+                    ).numpy().item()
+                _k = g.nodes['n3'].data['k%s' % suffix][position].detach(
+                    ).numpy().item()
+
+                _eq = Quantity(
+                    _eq,
+                    esp.units.ANGLE_UNIT,
+                ).value_in_unit(OPENMM_ANGLE_EQ_UNIT)
+
+                _k = 2.0 * Quantity(
+                    _k,
+                    esp.units.ANGLE_FORCE_CONSTANCE_UNIT,
+                ).value_in_unit(OPENMM_ANGLE_K_UNIT)
+
+                force.setAngleParameters(idx, idx0, idx1, idx2, _eq, _k)
+
+        if 'PeriodicTorsionForce' in name:
+            number_of_torsions = force.getNumTorsions()
+            blank_periodicity = False
+            if 'periodicity%s' % suffix not in g.nodes['n4'].data\
+                or 'phase%s' % suffix not in g.nodes['n4'].data:
+                blank_periodicity = True
+
+                g.nodes['n4'].data['periodicity%s' % suffix] = torch.arange(
+                    1, 6
+                )[None, :].repeat(
+                    g.heterograph.number_of_nodes('n4'), 1
+                )
+
+                g.nodes['n4'].data['phases%s' % suffix] = torch.zeros(
+                    g.heterograph.number_of_nodes('n4'),
+                    6
+                )
+
+            count_idx = 0
+            for idx in range(g.heterograph.number_of_nodes('n4')):
+                idx0 = g.nodes['n4'].data['idxs'][idx, 0].item()
+                idx1 = g.nodes['n4'].data['idxs'][idx, 1].item()
+                idx2 = g.nodes['n4'].data['idxs'][idx, 2].item()
+                idx3 = g.nodes['n4'].data['idxs'][idx, 3].item()
+
+                if idx0 < idx3:
```

Comment:
add comment about assumptions and intent of this condition

("assuming both (a,b,c,d) and (d,c,b,a) are listed for every torsion, only pick one of the orderings" ?)

Reply:
Resolved in https://github.com/choderalab/espaloma/pull/39/commits/81aa7f97fec08fc34b3c675b9cc951dbf01e7a3b